### ISSUE TITLE
insta complains falsy assertion failure if the text snapshot starts with `---`

### ISSUE THREAD

**Issue by Myriad-Dreamin** [mitsuhiko/insta#688]
### What happened?

I'm encountering https://github.com/Myriad-Dreamin/tinymist/issues/868 in insta 1.41.1. In short, the issue reports that insta complains assertion failures on nixpkgs's build targets `{x86_64,aarch64}-{linux,darwin}`. After debug for a while, I suspect it is because some of my text snapshots start with `---\n`, as shown in https://github.com/Myriad-Dreamin/tinymist/pull/903. However, I'm not 100% confident on my judgement because I cannot trigger the failure on my local PC (`{x86_64}-{windows,linux}`). Could you help me?

Some build log containing such failures: https://build.alpinelinux.org/buildlogs/build-edge-x86_64/testing/tinymist/tinymist-0.12.2-r1.log

### Reproduction steps

running `cargo test` might trigger the falsy assertion failure.

### Insta Version

1.41.1

### rustc Version

1.82

### What did you expect?

I expect to pass all tests.

**Comment by max-sixty** [comment]
Yes, it's possible with that exact string that we'd see odd behavior, since we strip it.

Do you have a minimal reproduction?

**Comment by max-sixty** [comment]
I also don't see test failures in the log you linked to? I see some warnings about legacy snapshots — which is very much expected on inline snapshots starting with `---\n` (insta's behavior used to be inconsistent between file & inline snapshots there). 

I will prioritize if we can find a case of inconsistency / backward-compat issues!

**Comment by max-sixty** [comment]
Ok, though I'm not sure what "_falsy_ warnings" are, I'm interpreting that at meaning there aren't bugs, let me know if that's not the case!
